tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1660,31 +1660,27 @@ Status StorageManager::load_array_metadata(
   const auto& array_metadata_to_load = array_dir.array_meta_uris();
 
   auto metadata_num = array_metadata_to_load.size();
-  std::vector<shared_ptr<Buffer>> metadata_buffs;
-  metadata_buffs.resize(metadata_num);
+  std::vector<shared_ptr<Tile>> metadata_tiles;
+  metadata_tiles.resize(metadata_num);
   auto status = parallel_for(compute_tp_, 0, metadata_num, [&](size_t m) {
     const auto& uri = array_metadata_to_load[m].uri_;
 
     auto&& [st, tile] = load_data_from_generic_tile(uri, 0, encryption_key);
     RETURN_NOT_OK(st);
 
-    metadata_buffs[m] = tdb::make_shared<Buffer>(HERE());
-    RETURN_NOT_OK(metadata_buffs[m]->realloc(tile->size()));
-    metadata_buffs[m]->set_size(tile->size());
-    RETURN_NOT_OK(
-        tile->read(metadata_buffs[m]->data(), 0, metadata_buffs[m]->size()));
+    metadata_tiles[m] = tdb::make_shared<Tile>(HERE(), std::move(*tile));
 
     return Status::Ok();
   });
   RETURN_NOT_OK(status);
 
   // Compute array metadata size for the statistics
   uint64_t meta_size = 0;
-  for (const auto& b : metadata_buffs)
-    meta_size += b->size();
+  for (const auto& t : metadata_tiles)
+    meta_size += t->size();
   stats_->add_counter("read_array_meta_size", meta_size);
 
-  *metadata = Metadata::deserialize(metadata_buffs);
+  *metadata = Metadata::deserialize(metadata_tiles);
 
   // Sets the loaded metadata URIs
   RETURN_NOT_OK(metadata->set_loaded_metadata_uris(array_metadata_to_load));
```

Comment:
We could probably convert set_loaded_metadata_uris to return void.

Reply:
done